### PR TITLE
fix(core): fix renku save with deleted files

### PR DESCRIPTION
**Pull request by Panaetius**
Fixes issue with getting staged paths and also dealing with deleted&staged files in `renku save`

### CODE REVIEW

File: renku/core/commands/save.py [hunk 1]
```diff
@@ -93,7 +93,13 @@ def repo_sync(repo, message=None, remote=None, paths=None):
     if paths:
         # commit uncommitted changes
         try:
-            repo.git.add(*paths)
+            staged_files = {d.a_path for d in repo.index.diff("HEAD")} if repo.head.is_valid() else set()
```
Comment (mohammad-sdsc) [inline renku/core/commands/save.py:96]: Should we call `git_unicode_unescape` on `d.a_path`? If so, then we should also call it on `paths` as well.

File: renku/core/management/git.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def modified_paths(self):
     def dirty_paths(self):
         """Get paths of dirty files in the repository."""
         repo_path = self.repo.working_dir
-        staged_files = self.repo.index.diff("HEAD") if self.repo.head.is_valid() else []
+        staged_files = [d.a_path for d in self.repo.index.diff("HEAD")] if self.repo.head.is_valid() else []
```
Comment (mohammad-sdsc) [inline renku/core/management/git.py:131]: Same about calling `git_unicode_unescape`.

**Review by mohammad-sdsc — COMMENTED** [review]
LGTM, just two comments about unicode file names.

File: renku/core/commands/save.py (hunk 1 above)
Comment (Panaetius) [reply renku/core/commands/save.py]: We only need to call it on paths that come from git, so yes about adding it here, but `paths` is supplied by us/the user, so there it's not necessary.

**Review by mohammad-sdsc — APPROVED** [review]
Thanks!

**Pull request merged**